functional/FunctionalTest.java

```diff
@@ -1074,11 +1074,17 @@ public static void copyObject_test7() throws Exception {
 
     copyConditions.setUnmodified(dateRepresentation);
 
+    ErrorResponseException expectedException = null;
+
     try {
       client.copyObject(bucketName, fileName, secondBucketName, copyConditions);
     } catch (ErrorResponseException e) {
       // File should not be copied as object was modified after date set in copyConditions.
-      ignore();
+      expectedException = e;
+    }
+
+    if (expectedException == null || !expectedException.errorResponse().code().equals("PreconditionFailed")) {
+      throw new Exception("[FAILED] copyObject_test7(): copying should fail with unsatisified precondition.");
     }
 
     client.removeObject(bucketName, fileName);
```

Comment:
You won't see any failure because removeObject never returns any error, with or without object exists or not.